presto-main/src/main/java/com/facebook/presto/operator/ExchangeOperator.java

```diff
@@ -106,6 +109,9 @@ public ExchangeOperator(
         this.exchangeClient = requireNonNull(exchangeClient, "exchangeClient is null");
         this.serde = requireNonNull(serde, "serde is null");
 
+        this.systemMemoryContext = operatorContext.localUserMemoryContext();
+        this.systemMemoryContext.setBytes(this.serde.getRetainedSizeInBytes());
```

Comment:
Why do you count the memory in the constructor? Would it be better to count it in getOutput() after the deserialization?